### PR TITLE
Fix paste-html heading problem. (#3320)

### PR DESCRIPTION
**Pull request by aykutkardas**
#### Is this adding or improving a _feature_ or fixing a _bug_?
Fix paste-html heading problem. (#3320)

#### What's the new behavior?

[Preview Link](https://imgur.com/a/Bhiez7Y)
Preview

#### How does this change work?

If the pasted line is empty, it deletes the current line and pastes the value in the clipboard. If the pasted line is not empty, combines the value in the clipboard with the current line.

#### Have you checked that...?

- [x] The new code matches the existing patterns and styles.
- [x] The tests pass with `yarn test`.
- [x] The linter passes with `yarn lint`. (Fix errors with `yarn fix`.)
- [x] The relevant examples still work. (Run examples with `yarn start`.)

#### Does this fix any issues or need any specific reviewers?

Fixes: #3320

### CODE REVIEW

**Comment by aykutkardas** [comment]
The original repo and the tests I run in my own repo give the same results.  @ianstormtaylor

File: packages/slate/src/transforms/text.ts [hunk 1]
```diff
@@ -350,6 +354,21 @@ export const TextTransforms = {
         voids,
       })
 
+      const [firstMatch] = matches
+      let firstMatchBlock
+
+      if (firstMatch) {
+        ;[firstMatchBlock] = firstMatch
```
Comment (CameronAckermanSEL) [inline packages/slate/src/transforms/text.ts]:
```suggestion
        [firstMatchBlock] = firstMatch
```

Comment (aykutkardas) [reply packages/slate/src/transforms/text.ts]: The reason for this is Prettier.

Comment (CameronAckermanSEL) [reply packages/slate/src/transforms/text.ts]: Ah ok thanks for pointing that out!

**Comment by CameronAckermanSEL** [comment]
Looking at this PR along with a few other pasting ones. I think because this has an observable behavior change with how merging of nodes works in pastes, it might call for some extra unit tests. I'll do a more in depth evaluation tonight or this weekend. Thanks so much for this PR, merging of blocks on paste is a tricky problem and has had some quirky behaviors in slate for a long time.

**Comment by CameronAckermanSEL** [comment]
I'm observing the test failure on my local clone of the repo. I think it's because `isMergeStart` is true in less situations, it's now firing this inline merging logic unexpectedly and failing. Might need to just add some short circuiting logic somewhere to ensure the inline logic isnt reached

```
  1 failing

  1) slate
       transforms
         insertFragment
           of-inlines
             block-empty :
     Error: Cannot find a descendant at path [0] in node: {"children":[],"operations":[{"type":"remove_node","path":[0],"node":{"children":[{"text":""}]}}],"selection":null,"marks":null}
      at Object.Node [as get] (../src/interfaces/node.ts:243:15)
      at Object.levels (../src/interfaces/node.ts:346:17)
      at levels.next (<anonymous>)
      at Object.levels (../src/interfaces/editor.ts:482:16)
      at levels.next (<anonymous>)
      at Object.Editor [as above] (../src/interfaces/editor.ts:90:16)
      at Object.void (../src/interfaces/editor.ts:1573:12)
      at fn (../src/transforms/node.ts:111:21)
      at Object.Editor [as withoutNormalizing] (../src/interfaces/editor.ts:1586:5)
      at Object.Transforms [as insertNodes] (../src/transforms/node.ts:30:5)
      at fn (../src/transforms/text.ts:379:7)
      at Object.Editor [as withoutNormalizing] (../src/interfaces/editor.ts:1586:5)
      at Object.insertFragment (../src/transforms/text.ts:200:5)
      at run (packages\slate\test\transforms\insertFragment\of-inlines/block-empty.js:7:14)
      at fn (packages\slate\test/index.js:43:5)
      at Context.<anonymous> (support/fixtures.js:46:11)
      at processImmediate (internal/timers.js:439:21)
```

This is at this line:

```
      const startRef = Editor.pathRef(
        editor,
        !isInlineStart || (isInlineStart && isInlineEnd)
          ? Path.next(inlinePath)
          : inlinePath
      )

      Transforms.insertNodes(editor, starts, {
        at: startRef.current!,
        match: n => Text.isText(n) || Editor.isInline(editor, n),
        mode: 'highest',
        voids,
      })
```

**Comment by alijardariasif** [comment]
On 26-Oct-2020 11:20 AM, "Aykut Kardaş" <notifications@github.com> wrote:

Closed #3494 <https://github.com/ianstormtaylor/slate/pull/3494>.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/ianstormtaylor/slate/pull/3494#event-3918950251>, or
unsubscribe

**Pull request closed without merging**